rules/library.bzl

```diff
@@ -456,22 +440,21 @@ def apple_library(name, library_tools = {}, export_private_headers = True, names
     # vfs_overlay(name = vfs_name, deps = deps)
     # internal_deps.append(vfs_name)
 
-    headermap_copts = []
-    headermap_copts.append("-I\"$(execpath :%s)\"" % private_hmap_name)
-    headermap_copts.append("-I\"$(execpath :%s)\"" % public_hmap_name)
-    headermap_copts.append("-I\"$(execpath :%s)\"" % private_angled_hmap_name)
-    if swift_sources:
-        headermap_copts.append("-iquote\"$(execpath :%s)\"" % swift_doublequote_hmap_name)
-    headermap_copts.append("-I.")
-    headermap_copts.append("-iquote\"$(execpath :%s)\"" % private_hmap_name)
+    _append_headermap_copts(private_hmap_name, "-I", objc_copts, swift_copts, cc_copts)
+    _append_headermap_copts(public_hmap_name, "-I", objc_copts, swift_copts, cc_copts)
+    _append_headermap_copts(private_angled_hmap_name, "-I", objc_copts, swift_copts, cc_copts)
+    objc_copts.append("-I.")
+    cc_copts.append("-I.")
+    swift_copts.extend(("-Xcc", "-I."))
+    _append_headermap_copts(private_hmap_name, "-iquote", objc_copts, swift_copts, cc_copts)
 
-    objc_copts += headermap_copts + [
+    objc_copts += [
         "-fmodules",
         "-fmodule-name=%s" % module_name,
         "-gmodules",
     ]
 
-    swift_copts += [j for i in ([["-Xcc", copt] for copt in headermap_copts]) for j in i] + [
+    swift_copts += [
         "-Xcc",
```

Comment:
it seems like you already added `-Xcc` on line 448.

Reply:
You need an `-Xcc` for every argument that you want passed to clang